tv_apps/smart-home/js/card_util.js

```diff
@@ -187,19 +188,18 @@
       var nameSpan = document.createElement('span');
       nameSpan.classList.add('name');
       this._localizeCardName(nameSpan, card);
-      cardButton.appendChild(nameSpan);
 
-      return cardButton;
+      var fragment = document.createDocumentFragment();
+      fragment.appendChild(cardButton);
+      fragment.appendChild(nameSpan);
+
+      return fragment;
     },
 
-    updateCardName: function(cardButton, card) {
-      var spans =
-           SharedUtils.nodeListToArray(cardButton.getElementsByTagName('span'));
-      spans.forEach(span => {
-        if (span.classList.contains('name')) {
-          this._localizeCardName(span, card);
-        }
-      });
+    updateCardName: function(nameElement, card) {
+      if (nameElement.classList.contains('name')) {
```

Comment:
If everyone call this API correctly, I suppose we don't need to verify its classList. (we may just need to check if it's `null`.)